R/groups.R

```diff
@@ -0,0 +1,184 @@
+#'     publisher or viewer.
+#'   * `created_time`: The timestamp (in RFC3339 format) when the user
+#'     was created in the Posit Connect server
+#'   * `updated_time`: The timestamp (in RFC3339 format) when the user
+#'     was last updated in the Posit Connect server
+#'   * `active_time`: The timestamp (in RFC3339 format) when the user
+#'     was last active on the Posit Connect server
+#'   * `confirmed`: When false, the created user must confirm their
+#'     account through an email. This feature is unique to password
+#'     authentication.
+#'   * `locked`: Whether or not the user is locked
+#'   * `guid`: The user's GUID, or unique identifier, in UUID RFC4122 format
+#'
+#' @details
+#' Please see https://docs.posit.co/connect/api/#get-/v1/groups/-group_guid-/members
+#' for more information.
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # get the first 20 groups
+#' groups <- get_groups(client)
+#'
+#' group_guid <- groups$guid[1]
+#'
+#' get_group_members(client, guid = group_guid)
+#' }
+#'
+#' @family groups functions
+#' @export
+get_group_members <- function(src, guid) {
+  validate_R6_class(src, "Connect")
+
+  res <- src$group_members(guid)
+
+  out <- parse_connectapi(res$results)
+
+  return(out)
+}
+
+#' Get content access permissions for a group or groups
+#'
+#' @param src The source object
+#' @param groups A data frame or tibble of groups
+#'
+#' @return
+#' A tibble with the following columns:
+
+#'   * `group_guid`: The group's GUID
+#'   * `group_name`: The group's name
+#'   * `content_guid`: The content item's GUID
+#'   * `content_name`: The content item's name
+#'   * `content_title`: The content item's title
+#'   * `access_type`: The access type of the content item ("all", "logged_in", or "acl")
+#'   * `role`: The access type that members of the group have to the
+#'     content item, "publisher" or "viewer".
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # Get a data frame of groups
+#' groups <- get_groups(client)
+#'
+#' # Get permissions for a single group by passing in the corresponding row.
+#' get_group_content(client, groups[1, ])
+#' dplyr::filter(groups, name = "research_scientists") %>%
+#'   get_group_content(client, groups = .)
+#'
+#' # Get permissions for all groups by passing in the entire groups data frame.
+#' get_group_content(client, groups)
+#' }
+#'
+#' @family groups functions
+#' @export
+get_group_content <- function(src, groups) {
+  validate_R6_class(src, "Connect")
+
+  purrr::pmap_dfr(
+    dplyr::select(groups, guid, name),
+    get_group_content_impl,
+    src = src
+  )
+}
+
+get_group_content_impl <- function(src, guid, name) {
+  validate_R6_class(src, "Connect")
+
+  res <- src$group_content(guid)
+  parsed <- parse_connectapi_typed(res, connectapi_ptypes$group_content)
+
+  dplyr::transmute(parsed,
+    group_guid = guid,
+    group_name = name,
+    content_guid,
```

Comment:
These lintr warnings (and R CMD check warnings) should be able to be removed if `.data$content_guid` is used. 

Related vignette: https://cran.r-project.org/web/packages/dplyr/vignettes/in-packages.html#data-masking-and-tidy-selection-notes

(This can be done another day if the package doesn't do it already)

Reply:
Thanks!

Reply:
Do you know why `lintr` doesn't seem to have a problem with some of the data-vars?

```r

  dplyr::transmute(parsed,
    group_guid = guid,
    group_name = name,
    .data$content_guid,
    .data$content_name,
    content_title,
    access_type,
    role = purrr::map_chr(
      .data$permissions,
      extract_role,
      principal_name = name
    )
  )
```

It seems fine with `content_title` and `access_type`. I'm adding `.data$` to them regardless for consistency & clarity, otherwise it looks like they are something different.

Reply:
Apparently

```
Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
i Please use `"guid"` instead of `.data$guid`
```